### ISSUE TITLE
[Bug]: Issue deploying aws_s3_directory_bucket with Terraform Cloud

### ISSUE THREAD

**Issue by ferrarijefferson** [hashicorp/terraform-provider-aws#34639]
### Terraform Core Version

1.0.6

### AWS Provider Version

5.28.0

### Affected Resource(s)

I have tried in a few ways to deploy the aws_s3_directory_bucket through Terraform Cloud, but in all tests, I get the same access denied result. What's strange is that when I run this locally, I don't encounter any failures.
The user I utilize for Terraform Cloud has full administrator access.

### Expected Behavior
```shell
Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # will be created
  + resource "aws_s3_directory_bucket" "s3_express" {
      + arn             = (known after apply)
      + bucket          = ""
      + data_redundancy = "SingleAvailabilityZone"
      + force_destroy   = true
      + id              = (known after apply)
      + type            = "Directory"

      + location {
          + name = "eun1-az1"
          + type = "AvailabilityZone"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

Creating...
Creation complete after 2s [id=]
```
### Actual Behavior

```shell
Terraform v1.0.6
on linux_amd64
Initializing plugins and modules...
Creating...
╷
│ Error: creating S3 Directory Bucket ()
│ 

│   10: resource "aws_s3_directory_bucket" "s3_express" {
│ 
│ operation error S3: CreateBucket, https response error StatusCode: 403,
│ RequestID:, HostID:
│, api error AccessDenied: Access Denied
╵
Operation failed: failed running terraform apply (exit 1)
```

### Terraform Configuration Files

```hcl
resource "aws_s3_directory_bucket" "s3_express" {
  bucket = "${data.aws_iam_account_alias.alias.account_alias}-${var.region}-test--${var.zone_id}--x-s3"

  location {
    name = var.zone_id
  }

  force_destroy = true
}
```
### Steps to Reproduce

I only get this error through Terraform Cloud.

### Would you like to implement a fix?

None

**Comment by ewbankkit** [comment]
@ferrarijefferson Thanks for raising this issue 👏.
I see `+ bucket          = ""` in the plan which suggest that no bucket name is specified, although it should be derived from the values in the configuration.

**Comment by ferrarijefferson** [comment]
Thank you very much for responding me @ewbankkit 
I apologize for forgetting to mention, I just removed the name because it contained information relevant to the company.

**Comment by ewbankkit** [comment]
Then it is likely to be an IAM permissions issue -- does the TFC role have the new `s3express` permissions: https://docs.aws.amazon.com/AmazonS3/latest/userguide/s3-express-security-iam.html#s3-express-security-iam-actions?

**Comment by ferrarijefferson** [comment]
> Then it is likely to be an IAM permissions issue -- does the TFC role have the new `s3express` permissions: https://docs.aws.amazon.com/AmazonS3/latest/userguide/s3-express-security-iam.html#s3-express-security-iam-actions?

thanks a million for that, and I'll take a look, but the user I'm running this test with has full access.
<img width="412" alt="image" src="https://github.com/hashicorp/terraform-provider-aws/assets/104831988/0b83d1f5-8b43-4660-9cf1-4c2755365ad1">

Either way, I will run some more tests next week.

**Comment by ferrarijefferson** [comment]
Hello, sorry for the delay in responding, I had a few days off.
After conducting some tests, I discovered what was happening.
AWS S3 express is available only in some AWS regions (which we already knew), so I forced my module to deploy only this service in the 'eu-north-1' region because my workspace provider is set up for the 'eu-west-1' region. Up to this point, everything was fine as we have already used this method with other resources like lambdas that we need to have in other regions. 
But especially for this resource, I had access denied for the deployment, so the solution was to create a new workspace with the default region set to 'eu-north-1'.

Anyway, thank you very much for your attention.
